.ci/pytorch/macos-build.sh

```diff
@@ -36,11 +36,11 @@ fi
 print_cmake_info
 if [[ ${BUILD_ENVIRONMENT} == *"distributed"* ]]; then
   # Needed for inductor benchmarks, as lots of HF networks make `torch.distribtued` calls
-  USE_DISTRIBUTED=1 USE_OPENMP=1 WERROR=1 python setup.py bdist_wheel
+  USE_DISTRIBUTED=1 USE_OPENMP=1 WERROR=1 python -m build --wheel --no-isolation
 else
   # Explicitly set USE_DISTRIBUTED=0 to align with the default build config on mac. This also serves as the sole CI config that tests
   # that building with USE_DISTRIBUTED=0 works at all. See https://github.com/pytorch/pytorch/issues/86448
-  USE_DISTRIBUTED=0 USE_OPENMP=1 MACOSX_DEPLOYMENT_TARGET=11.0 WERROR=1 BUILD_TEST=OFF USE_PYTORCH_METAL=1 python setup.py bdist_wheel --plat-name macosx_11_0_arm64
+  USE_DISTRIBUTED=0 USE_OPENMP=1 MACOSX_DEPLOYMENT_TARGET=11.0 WERROR=1 BUILD_TEST=OFF USE_PYTORCH_METAL=1 python -m build --wheel --no-isolation -C--build-option=--plat-name=macosx_11_0_arm64
```

Comment:
FYI, `build` 1.3.0 was released last week, which added a new `--config-json` option. That might be better than `--config-setting` / `-C`.

- https://github.com/pypa/build/releases/tag/1.3.0

Reply:
Thanks, that's an interesting new facility. It seems clear that we could move to `--config-json="{'--build-option': '--platform-name=macosx_11_0_arm64'}"`, but given the support in setuptools, I am afraid we can't go all the way to `--config-json="{'build-option': {'platform-name': 'macosx_11_0_arm64'}}"`. With the lack of support in older build versions, I think the best way forward is to stick with `--config-setting` for now and to reevaluate with a larger build system overhaul.